benchmark/scripts/Benchmark_Driver

```diff
@@ -178,14 +182,10 @@ class BenchmarkDriver(object):
 
         Returns the aggregated result of independent benchmark invocations.
         """
-        def merge_results(a, b):
-            a.merge(b)
-            return a
-
-        return reduce(merge_results,
-                      [self.run(test, measure_memory=True,
+        result_sets = [self.run(test, measure_memory=True,
                                 num_iters=1, quantile=20)
-                       for _ in range(self.args.independent_samples)])
+                       for _ in range(self.args.independent_samples)]
+        return reduce(lambda m, res: (m.merge(res), m)[1], result_sets)
```

Comment:
The original version of this seems less magical and easier for non-python folks to grok.